vim-patch:8.1.{1047,1101,1212},8.2.{742,749,769,874,2142,2189,2192}

janlazo opened the pull request:
I am including `test_signals.vim` for VimLeavePre test which doesn't run.

janlazo committed:
vim-patch:8.1.1047: WINCH signal is not tested

Problem:    WINCH signal is not tested.
Solution:   Add a test. (Dominique Pelle, closes vim/vim#4158)
https://github.com/vim/vim/commit/db77b84ac2b6373ae4200d47945fc6ca64337e31

janlazo committed:
vim-patch:8.1.1101: signals test may fail in the GUI

Problem:    Signals test may fail in the GUI.
Solution:   Skip the test for the GUI. (Yee Checng Chin, closes vim/vim#4202)
https://github.com/vim/vim/commit/690a905a0199f5ff733e5a12be9ba94ae6043ef4

janlazo committed:
vim-patch:8.1.1212: signal PWR is not tested

Problem:    Signal PWR is not tested.
Solution:   Test that PWR updates the swap file. (Dominique Pelle,
            closes vim/vim#4312)
https://github.com/vim/vim/commit/520e245237fa92f9022968a52eb418da6eade9b4

janlazo committed:
vim-patch:8.2.0742: handling of a TERM signal not tested

Problem:    Handling of a TERM signal not tested.
Solution:   Add a test for SIGTERM. (Dominique Pelle, closes vim/vim#6055)
https://github.com/vim/vim/commit/48a687148c4649f6f55b36a1f4111041e7207235

janlazo committed:
vim-patch:8.2.0749: TERM signal test fails on FreeBSD

Problem:    TERM signal test fails on FreeBSD.
Solution:   Do not check the messages, the may appear anywhere. (Dominique
            Pelle, closes vim/vim#6075)
https://github.com/vim/vim/commit/55ba4b844f1b0e44f0f2e1bd14d26e7ad2df9ffc

janlazo committed:
vim-patch:8.2.0769: VimLeavePre not triggered when Vim is terminated

Problem:    VimLeavePre not triggered when Vim is terminated.
Solution:   Unblock autocommands.
https://github.com/vim/vim/commit/129d6bf6b3d120b0a4c69e18b5e8602a84e352bf

janlazo committed:
vim-patch:8.2.0874: signals test is a bit flaky

Problem:    Signals test is a bit flaky.
Solution:   Flush the XautoOut file.  Delete files that may be left behind
            from a failure. (Dominique Pelle, closes vim/vim#6179)
https://github.com/vim/vim/commit/d14fd5285e491a39028c4b4722ddbe7c9dfa9bb2

janlazo committed:
vim-patch:8.2.2142: memory leak when heredoc is not terminated

Problem:    Memory leak when heredoc is not terminated.
Solution:   Free heredoc_trimmed.
https://github.com/vim/vim/commit/fffdf4754f4fd789c64596bdd9ea069725e63784

janlazo commented on src/nvim/testdir/test_signals.vim:
```diff
@@ -0,0 +1,140 @@
+  " 'lines' and 'columns' should have been restored
+  " after handing signal WINCH.
+  exe 'silent !kill -s WINCH ' .. getpid()
+  call WaitForAssert({-> assert_equal(old_lines, &lines)})
```
This always fails.

jamessan replied:
Maybe related to https://github.com/neovim/neovim/issues/11330.

janlazo replied:
Related but different issue. This test is suppose to not resize the terminal because `t_WS` option is set to empty string. Neovim does ignore `SIGWINCH` so `lines` and `columns` are not restored to the old values.

janlazo committed:
vim-patch:8.2.2189: cannot repeat a command that uses the small delete register

Problem:    Cannot repeat a command that uses the small delete register.
Solution:   Store the register name instead of the contents. (Christian
            Brabandt, closes vim/vim#7527)
https://github.com/vim/vim/commit/032a2d050b82b146d70d6ff714838ee62c07d8ad

N/A patches for version.c:

vim-patch:8.2.2192: Codecov on github actions fails

Problem:    Codecov on github actions fails.
Solution:   Revert to codecov script. (Ozaki Kiichi, closes vim/vim#7529)
https://github.com/vim/vim/commit/e5492609b3a153c631f1d600ecdef1af1c913bef

janlazo commented on src/nvim/testdir/test_signals.vim:
```diff
@@ -0,0 +1,140 @@
+
+" Test signal WINCH (window resize signal)
+func Test_signal_WINCH()
+  throw 'skipped: Nvim cannot avoid terminal resize'
```
Is there a way for the TUI to ignore specific signals?

janlazo commented:
Going to merge this. I'm assuming that ignoring signals will not be supported.
